### PR TITLE
Ft/zenko 20

### PR DESCRIPTION
**Pull request by anurag4DSB**
###  This pull request is for Zenko. A Prometheus server will pull metrics from S3 at the route: '/_/monitoring/metrics'. I created a new route handler for the same.

This change is required as all the components in Zenko are monitored by Prometheus. We need S3 too.

### CODE REVIEW

File: lib/utilities/monitoringHandler.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        cb(error, results);
+    });
+}
+
+
+function routeHandler(req, res, log, cb) {
+    if (req.method !== 'GET') {
+        return cb(errors.BadRequest, []);
+    }
+    client.collectDefaultMetrics();
```
Comment (NicolasT) [inline lib/utilities/monitoringHandler.js]: This should be only called once at application startup, not per-request?

File: lib/utilities/monitoringHandler.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    const promMetrics = client.register.metrics();
+    const contentLen = Buffer.byteLength(promMetrics, 'utf8');
+    res.setHeader('content-length', contentLen);
+    res.setHeader('content-type', 'application/json');
```
Comment (NicolasT) [inline lib/utilities/monitoringHandler.js]: Wrong MIME-type for the response. There's a constant defined I believe in prom-client, see https://github.com/siimon/prom-client#utilites

File: lib/utilities/monitoringHandler.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    res.end(promMetrics);
+    return undefined;
```
Comment (NicolasT) [inline lib/utilities/monitoringHandler.js]: Shouldn't be required.

File: lib/utilities/monitoringHandler.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+function routeHandler(req, res, log, cb) {
+    if (req.method !== 'GET') {
+        return cb(errors.BadRequest, []);
+    }
+    client.collectDefaultMetrics();
+    const promMetrics = client.register.metrics();
+    const contentLen = Buffer.byteLength(promMetrics, 'utf8');
+    res.setHeader('content-length', contentLen);
```
Comment (NicolasT) [inline lib/utilities/monitoringHandler.js]: Shouldn't be required, and maybe even set by the HTTP server module? Anyway, calculating the length can be somewhat expensive without a need.

File: lib/utilities/monitoringHandler.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    res.writeHead(statusCode, { 'Content-Type': 'application/json' });
+    res.write(JSON.stringify(results));
+    res.end(() => {
+        cb(error, results);
+    });
+}
+
+
+function routeHandler(req, res, log, cb) {
```
Comment (NicolasT) [inline lib/utilities/monitoringHandler.js]: In some control flows, `cb` is not called, which is slightly odd.

File: lib/utilities/monitoringHandler.js (hunk 3 above)
Comment (anurag4DSB) [reply lib/utilities/monitoringHandler.js]: This is to satisfy CI. Es-lint expects me to return

File: lib/utilities/monitoringHandler.js (hunk 4 above)
Comment (anurag4DSB) [reply lib/utilities/monitoringHandler.js]: This was suggested by the lead. It is required.

File: lib/utilities/monitoringHandler.js (hunk 1 above)
Comment (anurag4DSB) [reply lib/utilities/monitoringHandler.js]: So what do you suggest the timeout value to be. I thought we can control it better with the server controlling the timeout.

File: lib/utilities/monitoringHandler.js (hunk 5 above)
Comment (anurag4DSB) [reply lib/utilities/monitoringHandler.js]: As advised I am following the healthcheckHandler

File: lib/utilities/monitoringHandler.js (hunk 2 above)
Comment (anurag4DSB) [reply lib/utilities/monitoringHandler.js]: If I check the type, its string.
So it should be text/plain.
Is that okay?

File: lib/utilities/monitoringHandler.js (hunk 1 above)
Comment (anurag4DSB) [reply lib/utilities/monitoringHandler.js]: Also, existing intervals are automatically cleared when calling collectDefaultMetrics.

File: lib/utilities/monitoringHandler.js (hunk 2 above)
Comment (NicolasT) [reply lib/utilities/monitoringHandler.js]: > The content-type prometheus expects is also exported as a constant, both on the register and from the main file of this project, called contentType.

File: lib/utilities/monitoringHandler.js (hunk 1 above)
Comment (NicolasT) [reply lib/utilities/monitoringHandler.js]: I'd just set the interval to 5s or so, we won't be using a higher sampling rate, and not call it for every server request.

File: lib/utilities/monitoringHandler.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+function writeResponse(res, error, log, results, cb) {
+    let statusCode = 200;
+    if (error) {
+        if (Number.isInteger(error.code)) {
+            statusCode = error.code;
+        } else {
+            statusCode = 500;
+        }
+    }
+    res.writeHead(statusCode, { 'Content-Type': 'text/plain' });
```
Comment (NicolasT) [inline lib/utilities/monitoringHandler.js]: This is likely `application/json`, looking at what's below?

**Pull request merged**